### ISSUE TITLE
Tests not passing locally?

### ISSUE THREAD

**Issue by davidweichiang** [diprism/fggs#185]
When I run `make test` using Python 3.9 and PyTorch 1.9, I currently get:
```
...
  File "/Users/dchiang/Research/diprism/python/_build/pip_packages/lib/python3.9/site-packages/torch_semiring_einsum/equation.py", line 215, in get_bytes_per_element
    return get_bits_per_element(dtype) // 8
  File "/Users/dchiang/Research/diprism/python/_build/pip_packages/lib/python3.9/site-packages/torch_semiring_einsum/equation.py", line 212, in get_bits_per_element
    return torch.iinfo(dtype).bits
TypeError: torch.bool is not supported by torch.iinfo
```

When @grantexley uses Python 3.11.8 and PyTorch 2.4.0, he gets a different error which he will post momentarily.

**Comment by grantexley** [comment]
This is the error:

FAIL: test_infinite_gradient (test.test_sum_product.TestSumProduct.test_infinite_gradient)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/grantmacbook/fggs/test/test_sum_product.py", line 157, in test_infinite_gradient
    self.assertEqual(fgg.factors['fac1'].weights.grad.item(), math.inf)
AssertionError: 0.0 != inf

**Comment by ccshan** [comment]
@davidweichiang's problem seems caused by using torch_semiring_einsum without https://github.com/bdusell/semiring-einsum/pull/34 merged.

**Comment by ccshan** [comment]
I'm not getting @grantexley's problem using Python 3.12.6 and PyTorch 2.4.1+cu121. This is a long shot but perhaps try upgrading PyTorch??

**Comment by davidweichiang** [comment]
Does the version in pip have the necessary patches? If so, we can change the requirements to use the version in pip instead of GitHub.

**Comment by davidweichiang** [comment]
Confirmed that the semiring-einsum in pip works. Will make a PR.

@grantexley tried with Python 3.12.3 and PyTorch 2.4.1 (no CUDA) and still gets the same error.
@ccshan were you testing with CUDA?

**Comment by davidweichiang** [comment]
I get the same error (Mac, no CUDA, Homebrew, Python 3.12.5, PyTorch 2.4.1).

**Comment by davidweichiang** [comment]
Python 3.9.7 and PyTorch 2.2.2 is okay.

**Comment by davidweichiang** [comment]
I've traced the problem to the following line:

https://github.com/diprism/fggs/blob/7da35529f84cfd35012c05a336cc7f67cad4595b/fggs/semirings.py#L160

The comment says that if `a` has an infinite eigenvalue, then the solution will have `-0.0` in it somewhere. And in some PyTorch versions, this is not true; it has `0.0` instead.

Here's the minimal program that shows the difference:
```
import torch
a = torch.tensor([[-torch.inf]])
b = torch.tensor([1.])
print(torch.linalg.solve(a, b))
```

**Comment by davidweichiang** [comment]
Yes, I really did write code that depended on the difference between `0.0` and `-0.0`. Sorry.

**Comment by davidweichiang** [comment]
This shouldn't be closed until #189 is merged
